peakina/io/s3/s3_utils.py

```diff
@@ -71,7 +71,15 @@ def _s3_open_file_with_retries(fs: s3fs.S3FileSystem, path: str, retries: int) -
 def s3_open(url: str, *, client_kwargs: dict[str, Any] | None = None) -> IO[bytes]:
     """opens a s3 url and returns a file-like object"""
     access_key, secret, bucketname, objectname = parse_s3_url(url)
-    fs = s3fs.S3FileSystem(key=access_key, secret=secret, client_kwargs=client_kwargs)
+
+    if client_kwargs is not None and "session_token" in client_kwargs:  # pragma: no cover
+        session_token = client_kwargs.pop("session_token")
+        fs = s3fs.S3FileSystem(
+            key=access_key, secret=secret, token=session_token, client_kwargs=client_kwargs
```

Comment:
```suggestion
            key=access_key, secret=secret, token=session_token, client_kwargs=client_kwargs or None
```
An empty dict might be passed here, which is now what we want.